BUILD.gn

```diff
@@ -0,0 +1,507 @@
+    "runtime/renderer/android/xwalk_permission_client.h",
+    "runtime/renderer/android/xwalk_render_thread_observer.cc",
+    "runtime/renderer/android/xwalk_render_thread_observer.h",
+    "runtime/renderer/android/xwalk_render_view_ext.cc",
+    "runtime/renderer/android/xwalk_render_view_ext.h",
+    "runtime/renderer/isolated_file_system.cc",
+    "runtime/renderer/isolated_file_system.h",
+    "runtime/renderer/xwalk_content_renderer_client.cc",
+    "runtime/renderer/xwalk_content_renderer_client.h",
+    "runtime/renderer/xwalk_render_thread_observer_generic.cc",
+    "runtime/renderer/xwalk_render_thread_observer_generic.h",
+  ]
+  if (is_linux && !is_component_build && is_component_ffmpeg) {
+    # Set RPath for executable xwalk, xwalk_browsertest and xwalk_unittest.
+    public_configs = [ "//build/config/gcc:rpath_for_built_shared_libraries" ]
+  }
+
+  # TODO(heke123): Generate header files into gen/xwalk/grit by putting the
+  # files in the right places in grit. Clean below include_dirs.
+  include_dirs = [
+    "$root_gen_dir/xwalk",
+    "$root_gen_dir/xwalk/application",
+    "$root_gen_dir/xwalk/resources",
+    "$root_gen_dir/xwalk/sysapps",
+  ]
+  defines = [
+    "XWALK_VERSION=\"$xwalk_version\"",
+    "CHROME_VERSION=\"$chrome_version\"",
+  ]
+  deps = [
+    ":generate_upstream_blink_version",
+    "//base",
+    "//base:base_static",
+    "//base:i18n",
+    "//base/third_party/dynamic_annotations",
+    "//cc",
+    "//components/autofill/content/browser",
+    "//components/autofill/content/renderer",
+    "//components/autofill/core/browser",
+    "//components/cdm/renderer",
+    "//components/devtools_http_handler",
+    "//components/resources:components_resources",
+    "//components/strings",
+    "//components/url_formatter",
+    "//components/user_prefs",
+    "//components/visitedlink/browser",
+    "//components/visitedlink/renderer",
+    "//content",
+    "//content/public/app:both",
+    "//content/public/browser",
+    "//content/public/child",
+    "//content/public/common",
+    "//content/public/utility",
+    "//gin",
+    "//ipc",
+    "//media",
+    "//net",
+    "//net:net_resources",
+    "//skia",
+    "//storage/browser",
+    "//storage/common",
+    "//third_party/WebKit/public:blink",
+    "//third_party/boringssl",
+    "//ui/base",
+    "//ui/gl",
+    "//ui/shell_dialogs",
+    "//ui/snapshot",
+    "//url",
+    "//v8",
+    "//xwalk/application:xwalk_application_lib",
+    "//xwalk/experimental",
+    "//xwalk/extensions",
+    "//xwalk/resources:xwalk_pak",
+    "//xwalk/resources:xwalk_resources",
+    "//xwalk/resources:xwalk_resources_100_percent_pak",
+    "//xwalk/resources:xwalk_resources_200_percent_pak",
+    "//xwalk/resources:xwalk_resources_300_percent_pak",
+    "//xwalk/resources:xwalk_strings",
+    "//xwalk/sysapps",
+  ]
+
+  if (is_linux) {
+    deps += [ "//xwalk/resources:xwalk_locales" ]
+  }
+
+  # TODO: Add WebNotification for Linux
```

Comment:
The usual style from Chromium (which we follow) is that TODOs and FIXMEs should always have someone responsible for it, so this should be "TODO(heke123): ...".